### PR TITLE
return types for ensime-edit-definition

### LINKED ISSUE #358: ensime-edit-definition should return nil/t
with this

```elisp
(defun ensime-edit-definition (&optional where)
  "Lookup the definition of the name at point."
  (interactive)
  (let* ((info (ensime-rpc-symbol-at-point))
         (pos (ensime-symbol-decl-pos info)))
    (if (ensime-pos-valid-local-p pos)
        (progn
          (ensime-push-definition-stack)
          (ensime-goto-source-location pos where)
          t)
      (message "Sorry, no definition found.")
      nil)))
```

I can do

```elisp
(defun ensime-edit-definition-with-fallback ()
  "Variant of `ensime-edit-definition' with ctags if ENSIME is not available."
  (interactive)
  (unless (and (ensime-connection-or-nil)
               (ensime-edit-definition))
    (projectile-find-tag)))
```

which is rather convenient.

### PR DESCRIPTION
**Pull request by fommil**
close #358 

@aemoncannon @greenrd whoever is around. This is a tiny one.

### CODE REVIEW

**Comment by fommil** [comment]
@greenrd `! Ping`

**Comment by aemoncannon** [comment]
lgtm

File: ensime-editor.el [hunk 1]
```diff
@@ -271,17 +271,16 @@
...
     (if (ensime-pos-valid-local-p pos)
-	(progn
-	  (ensime-push-definition-stack)
-	  (ensime-goto-source-location pos where))
-      (message "Sorry, no definition found."))))
-
+        (progn
+          (ensime-push-definition-stack)
+          (ensime-goto-source-location pos where))
+      (not (message "Sorry, ENSIME couldn't find the definition.")))))
```
Comment (aemoncannon) [inline ensime-editor.el]: What's this line supposed to do?

Comment (fommil) [inline ensime-editor.el]: Just legacy compatibility, we always print when we don't find anything. But message returns the message, so we null it.

Comment (aemoncannon) [inline ensime-editor.el]: Ah hah. Thanks : )

**Pull request merged**